code/go/ecs/file.go

```diff
@@ -34,9 +34,19 @@ type File struct {
 	Name string `ecs:"name"`
 
 	// Directory where the file is located.
+	// On platforms with drive letters, the drive letter should not be part of
+	// the directory field. For example, "C:\\Program Files\\Foo" should be
+	// parsed as `file.directory` "Program Files\\Foo" and `drive.letter` "C".
 	Directory string `ecs:"directory"`
 
-	// Full path to the file.
+	// Drive letter where the file is located.
+	Drive string `ecs:"drive"`
+
+	// Full path to the file, including the file name.
+	// On platforms with drive letters, the drive letter should not be part of
+	// the `file.path` field. For example, "C:\\Program Files\\Foo\\Bar.exe"
```

Comment:
I'm not sure if this is a good idea.
This could get really messy, and the format should match `process.path` and other fields that might be worth joining on.

We could add `relative_path` if we wanted to, but that ends up meaning we index even more data.

Reply:
I approached it this way in order to address detections that need to be drive-letter independent.

But I think you're right about pivoting between process.executable, and frankly, user expectations.

I'll ponder this a bit (aka let other people chime in), but I'm thinking I'll go with your suggestion here, and leave the drive letter in the path.